tests/ui/expr/weird-exprs.rs

```diff
@@ -271,6 +272,24 @@ fn fish_fight() {
     tug_o_war(<T>::_____________::<T>);
 }
 
+// all 48 keywords in 300 characters
+#[allow(warnings)]
+mod x {
+    pub(super) struct X;
+    use Ok;
+    impl X {
+        pub(in crate) async fn x(self: Self, x: &'static &'_ dyn for<> Fn()) where {
+            unsafe extern { safe fn x(); }
+            macro_rules! x { () => {}; }
+            if 'x: loop {
+                return match while let true = break 'x false { continue } {
+                    ref x => { &raw mut x; async { const { enum A {} } }.await as () },
+                };
+            } { type x = X; } else { move || { trait x { }; union B { x: () } }; }
+        }
+    }
+}
```

Comment:
Suggestion: actually, can you please move this to a new test under `tests/ui/keyword`, this is not really exercising weird _expressions_ per se, the intention is to have a dense set of keywords.